app/policies/agent/team_policy.rb

```diff
@@ -4,20 +4,30 @@ def initialize(current_agent, team)
     @team = team
   end
 
-  def update?
-    self.class.allowed_to_manage_teams_in?(@team.territory, @current_agent)
-  end
-
   def self.allowed_to_manage_teams_in?(territory, agent)
     agent.access_rights_for_territory(territory)&.allow_to_manage_teams?
   end
 
+  def update?
+    allowed_to_manage_teams? && agents_in_team_territory?
+  end
+
   alias new? update?
   alias create? update?
   alias edit? update?
   alias destroy? update?
   alias versions? update?
 
+  private
+
+  def allowed_to_manage_teams?
+    self.class.allowed_to_manage_teams_in?(@team.territory, @current_agent)
+  end
+
+  def agents_in_team_territory?
```

Comment:
nitpick de nommage, à prendre ou à laisser sans soucis ! 

```suggestion
  def all_agents_belong_to_team_territory?
```

Reply:
Excellente suggestion, merci ! :ok_hand: